src/core/tsi/private_key_offload_util.h

```diff
@@ -0,0 +1,89 @@
+//
+//
+// Copyright 2025 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#ifndef GRPC_SRC_CORE_TSI_PRIVATE_KEY_OFFLOAD_UTIL_H
+#define GRPC_SRC_CORE_TSI_PRIVATE_KEY_OFFLOAD_UTIL_H
+
+#include <grpc/private_key_signer.h>
+#include <openssl/ssl.h>
+
+#include <string>
+
+#include "src/core/tsi/transport_security_interface.h"
+#include "absl/status/statusor.h"
+
+namespace grpc_core {
+
+// State associated with an SSL object for async private key operations.
+struct TlsPrivateKeyOffloadContext {
+  enum SignatureStatus {
+    // The signature operation has not yet started.
+    kNotStarted,
+    // The signature operation has been initiated.
+    kStarted,
+    // The signature operation is currently in progress waiting for an
+    // asynchronous operation.
+    kInProgressAsync,
+    // The signature operation has completed, and the signed data is available
+    // on the cached context.
+    kSignatureCompleted,
+    // The entire private key offload process for this signature is finished.
+    kFinished,
+  };
+
+  absl::StatusOr<std::string> signed_bytes;
```

Comment:
nit: Please add a comment clarifying when this is populated vs. when this is an error.